spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java

```diff
@@ -407,6 +407,11 @@ protected void onInit() {
 		 */
 		this.lockRegistrySet = true;
 		this.forceReleaseProcessor = createGroupTimeoutProcessor();
+
+		if (this.releaseStrategy instanceof GroupConditionProvider) {
+			this.messageStore.setConditionSupplier(
+					((GroupConditionProvider) this.releaseStrategy).getGroupConditionSupplier());
+		}
```

Comment:
I don't think this will work - while we recommend a different store for each aggregator, we don't enforce it.